java-version/accelerator.yaml

```diff
@@ -14,23 +14,49 @@ accelerator:
     required: true
 
 engine:
+  let:
+    - name: workloadJavaVersion
+      expression: "#javaVersion == '1.8' ? '8' : #javaVersion"
   merge:
+    # Maven
     - include: [ "pom.xml" ]
       chain:
       - type: ReplaceText
         regex:
           pattern: "<java.version>.*<"
           with: "'<java.version>' + #javaVersion + '<'"
+
+    # Gradle Groovy DSL
     - include: [ "build.gradle" ]
       chain:
       - type: ReplaceText
         regex:
           pattern: "sourceCompatibility = .*"
           with: "'sourceCompatibility = ''' + #javaVersion + ''''"
-    - include: [ "config/workload.yaml" ]
+
+    # Gradle Kotlin DSL
+    - include: [ "build.gradle.kts" ]
       chain:
       - type: ReplaceText
-        condition: "#javaVersion == '17'"
-        substitutions:
-          - text: "spec:"
-            with: "'spec:\n  build:\n    env:\n    - name: BP_JVM_VERSION\n      value: \"17\"'"
+        regex:
+          pattern: "(?<unmodified>JavaVersion\\.VERSION_)(\\d*)"
+          with: "'${unmodified}' + #javaVersion.replace('.', '_')"
+
+    # workload.yaml
+    # Covers a case when BP_JVM_VERSION is already defined in the workload.yaml
+    - include: [ "config/workload.yaml" ]
+      chain:
+      - type: OpenRewriteRecipe
+        recipe: org.openrewrite.yaml.ChangeValue
+        options:
+          oldKeyPath: '"$.spec.build.env[?(@.name == ''BP_JVM_VERSION'')].value"'
+          value: "#workloadJavaVersion"
+    # Covers a case when BP_JVM_VERSION is not defined yet in the workload.yaml
+    - condition: "!#files.contentsOf('config/workload.yaml').get().contains('- name: BP_JVM_VERSION')"
```

Comment:
This condition fails when using this fragment with the existing `hello-fun` sample accelerator. The error is `java.lang.IllegalArgumentException: Unable to find file 'config/workload.yaml'`

Reply:
True, it seems that the condition is elevated before the host accelerator did the path replacement. In order to make it more robust removed this condition and addition of BP_JVM_VERSION and updated the hello-fun accelerator to declare the variable so that it will be changed by this Java Fragment.

Reply:
I would prefer to have it the other way around and not insert overrides of default build parameters unless necessary. IMO it would be better to require that the host accelerator does not specify `BP_JVM_VERSION` and then only insert it when the user requests a Java version of `17` that does not work with the default.

If OpenRewrite adds a recipe that lets you update or add value at a key, then we can relax this requirement.

I think it is fine to use the `MergeYaml` recipe for inserting the `BP_JVM_VERSION` entry.

Reply:
On second look there is no real clean way of inserting this YAML snippet so we should fall back on requiring a  `BP_JVM_VERSION` entry that can me modified with an OpenRewrite recipe.